dispatcher/dispatcher.go

```diff
@@ -155,8 +183,11 @@ Loop:
 			go func(job jobqueue.Job) {
 				defer wg.Done()
 				defer func() { <-d.sem }()
-				rslt := d.worker.Work(job)
-				d.jobqueue.Complete(job, rslt)
+				err := d.limiter.Wait(ctx)
```

Comment:
https://pkg.go.dev/golang.org/x/time/rate#Limiter.WaitN
> It returns an error if n exceeds the Limiter's burst size, the Context is canceled, or the expected wait time exceeds the Context's Deadline.

Reply:
Yes, the order is preserved. [`Wait` internally uses `reserveN` API](https://cs.opensource.google/go/x/time/+/master:rate/rate.go;l=245), which returns delay taking early reservations into consideration. So, for example, if the first goroutine waited for 2 secs, the next would wait for 4 secs.